### PR TITLE
 Add unsafe modifier when necessary 

### PR DESCRIPTION
**Pull request by jnm2**

### CODE REVIEW

File: src/PublicApiGeneratorTests/Field_modifiers.cs [hunk 1]
```diff
@@ -42,14 +42,31 @@ public void Include_const_fields()
...
+        public void Include_unsafe_fields()
+        {
+            // Have to include the ctor - I can't figure out how to hide it
+            // when values are initialized
+            AssertPublicApi<ClassWithUnsafeFields>(
+                @"namespace PublicApiGeneratorTests.Examples
+{
+    public class ClassWithUnsafeFields
+    {
+        protected unsafe System.Void* UnsafeProtectedField;
```
Comment (sungam3r) [inline src/PublicApiGeneratorTests/Field_modifiers.cs]: `System.Void` should be changed later to `void` by #115

Comment (jnm2) [reply src/PublicApiGeneratorTests/Field_modifiers.cs]: Btw I'm happy to wait till your PR is merged and rebase mine so that these PRs aren't a burden.

Comment (sungam3r) [reply src/PublicApiGeneratorTests/Field_modifiers.cs]: #115 has grown much wider than I expected

Comment (jnm2) [reply src/PublicApiGeneratorTests/Field_modifiers.cs]: I rebased this branch on master after #115 was merged. Apparently it did not change to `void`.

Comment (sungam3r) [reply src/PublicApiGeneratorTests/Field_modifiers.cs]: I was wrong. See jnm2/ApiApprover#1

**Comment by jnm2** [comment]
I fast-forwarded to @sungam3r's commit locally and then force-pushed. (I appreciate the help, Ivan!)

Ready for review.

File: src/PublicApiGeneratorTests/PublicApiGeneratorTests.csproj [hunk 2]
```diff
@@ -9,6 +9,9 @@
     IDE0060 Remove unused parameter 'name' if it is not part of a shipped public API
     IDE1006 Naming rule violation: These words must begin with upper case characters
     -->
+    <NoWarn>$(NoWarn);CS0067</NoWarn>
```
Comment (sungam3r) [inline src/PublicApiGeneratorTests/PublicApiGeneratorTests.csproj]: Merge this with NoWarns above.

File: src/PublicApiGeneratorTests/PublicApiGeneratorTests.csproj [hunk 3]
```diff
@@ -9,6 +9,9 @@
...
+    <LangVersion>latest</LangVersion>
```
Comment (sungam3r) [inline src/PublicApiGeneratorTests/PublicApiGeneratorTests.csproj]: This already exists in Directory.Build.props

File: src/PublicApiGeneratorTests/PublicApiGeneratorTests.csproj (hunk 2 above)
Comment (jnm2) [reply src/PublicApiGeneratorTests/PublicApiGeneratorTests.csproj]: I'm confused. I'm 100% positive that I did not type this. Maybe I clicked something in VS by mistake.

Comment (jnm2) [reply src/PublicApiGeneratorTests/PublicApiGeneratorTests.csproj]: Could have been git automerging when I rebased.

**Comment by jnm2** [comment]
Thanks, fixed!

**Comment by sungam3r** [comment]
It will be good to add tests for different primitive types - `int*`, `byte*` etc., not ony `void*`.

**Comment by danielmarbach** [comment]
Replaced by https://github.com/ApiApprover/ApiApprover/pull/120

Added a few more test cases like mentioned by @sungam3r

**Comment by jnm2** [comment]
Oops, I forgot to do this last night. Thanks for taking care of it.

**Pull request closed without merging**